File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/strategy/transport/Bike.java
package com.bigboxcode.designpattern.strategy.transport;

import com.bigboxcode.Transport;

public class Bike implements Transport {

    @Override
    public void start() {
        System.out.println("Bike started");
    }

    @Override
    public void stop() {
        System.out.println("Bike stopped");
    }

    @Override
    public void repair() {
        System.out.println("Bike repair");
    }

    @Override
    public void getInfo() {
        System.out.println("Transport type: Bike");
        System.out.println("Number of wheels: 2");
        System.out.println("Average Weight: 700 Pounds");
    }

    @Override
    public void operate() {
        System.out.println("Riding Bike ............");
    }
}

File: design-patterns/php/strategy/transport/Bike.php
<?php
// Bike.php

namespace BigBoxCode\DesignPattern\Strategy\Transport;


class Bike implements Transport {
    public function start(): void {
        echo "Bike started\n";
    }
    
    public function stop(): void {
        echo "Bike stopped\n";
    }

    public function repair(): void {
        echo "Bike repair\n";
    }

    public function getInfo(): void {
        echo "Transport type: Bike\n";
        echo "Number of wheels: 2\n";
        echo "Average Weight: 700 Pounds\n";
    }

    public function operate(): void {
        echo "Riding Bike ............\n";
    }
}